### PR TITLE
Honor snippet capability when using the extract function assist

### PR DESCRIPTION
**Pull request by Arthamys**
This fixes issue #7793

### CODE REVIEW

File: crates/ide_assists/src/handlers/extract_function.rs [hunk 1]
```diff
@@ -112,7 +112,12 @@ pub(crate) fn extract_function(acc: &mut Assists, ctx: &AssistContext) -> Option
 
             let fn_def = format_function(ctx, module, &fun, old_indent, new_indent);
             let insert_offset = insert_after.text_range().end();
-            builder.insert(insert_offset, fn_def);
+            match ctx.config.snippet_cap {
+                Some(_) => {
+                    builder.insert_snippet(ctx.config.snippet_cap.unwrap(), insert_offset, fn_def)
+                }
```
Comment (Veykril) [inline crates/ide_assists/src/handlers/extract_function.rs:118]:
```suggestion
            match ctx.config.snippet_cap {
                Some(cap) => builder.insert_snippet(cap, insert_offset, fn_def),
```
Didn't notice that earlier sorry

Comment (Arthamys) [reply crates/ide_assists/src/handlers/extract_function.rs]: Yeah you're right, didn't notice the anti-pattern, thanks for catching it!

**Comment by bjorn3** [comment]
@Arthamys Something for next time: If you write `fixes issue #...` instead of `fixes #...`, github doesn't recognize it.

**Pull request merged**